app/js/lib/reducers/form_content.js

```diff
@@ -25,6 +35,32 @@ function stateContainsUnknownKeys( state ) {
 	return !_.isEmpty( getInvalidKeys( state ) );
 }
 
+function clearFieldsIfAddressTypeChanges( newState, payload ) {
+	if ( payload.contentName !== 'addressType'  ) {
+		return;
+	}
+	switch ( payload.value ) {
+		case 'privat':
```

Comment:
Random comment: so those things are not going to be switched to some English equivalents? BTW: if I remember right, the old application used `person` instead of `privat`, along with other address types `firma` and `anonym` which I see are still used below.
Not that the old application should be seen as a reference of what is good. Just pointing this out as my guess is to have those values here "compatible" with those from the old app.